src/libraries/Microsoft.Extensions.Diagnostics/src/Metrics/DebugConsoleMetricListener.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.Extensions.Diagnostics.Metrics
+{
+    internal sealed class DebugConsoleMetricListener : IMetricsListener, IDisposable
+    {
+        private readonly Timer _timer;
+        internal TextWriter _textWriter = Console.Out;
+        private IObservableInstrumentsSource? _source;
+
+        public DebugConsoleMetricListener()
+        {
+            _timer = new Timer(OnTimer, null, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(1));
```

Comment:
>_timer [](http://example.com/codeflow?start=12&length=6)

would it be better to create the timer only when encountering first observable instrument?